### PR TITLE
Update key generation to provide fixed signatures and encoded checksums (for improved security tooling)

### LINKED ISSUE #7662: Consider updating key generation to increase identifiability (and glean security value from improved tooling)
#### What problem would the feature you're requesting solve? Please describe.
Inadvertent plaintext exposure of secrets is a significant industry problem. Many modern security models are therefore updating their key generation logic to encode fixed signatures and checksums in secret data. This approach allows various scanners (such as Microsoft Defender and [GitHub Token Scanning](https://github.blog/2018-10-17-behind-the-scenes-of-github-token-scanning/), to efficiently detect/block exposures.

Without improvement of this kind, tools must either generate large #'s of false positives (which prevent meaningful engineer review and security response) or favor approaches that generate false negatives (which leaves projects exposed to security risk).

#### Describe the solution you'd like
Encode a fixed signature (expressed in the base64-encoded secret) in Azure function secrets. Encode a checksum in the final four bytes of the decoded data. Provide both mechanisms without compromising required security (bits of entropy) in the generated secret. 

GitHub's post on its [new authentication format](https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/) provides a good description of approach. Note that Azure DevOps, NuGet, and NPM also have implemented similar technique to provide additional security value.

#### Describe alternatives you've considered
Some providers have exclusively utilized a fixed signature (often a prefix) in the encoded value. This approach is subject to collisions, however, if there is a single (unrelated) data generator that duplicates the signature.

### PR DESCRIPTION
**Pull request by fabiocav**
Mirror PR for #8071 

### Issue describing the changes in this PR
Resolves #7662

### Pull request checklist

* [X] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [X] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [X] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [X] I have added all required tests (Unit tests, E2E tests)

@liliankasem @fabiocav @brettsam @mathewc 

Will send some additional context offline.

### CODE REVIEW

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json [hunk 1]
```diff
@@ -608,7 +609,7 @@
           }
         }
       },
-      "Microsoft.Azure.Functions.JavaWorker/2.1.0-SNAPSHOT": {},
+      "Microsoft.Azure.Functions.JavaWorker/2.0.0": {},
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:612]: Why the change here?

**Review by mathewc — APPROVED** [review]

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json (hunk 1 above)
Comment (fabiocav) [reply test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:612]: Likely a result of the rebase/merge, but this doesn't have any effect on validation.

**Comment by fabiocav** [comment]
@michaelcfanning merging this. Will follow up with an update to runtime assemblies.
